runtime/interpreter/encode.go

```diff
@@ -843,7 +848,7 @@ func (e *Encoder) encodeDictionaryValue(
 		return err
 	}
 
-	keysPath := append(path[:], dictionaryKeyPathPrefix)
+	keysPath := append(path, dictionaryKeyPathPrefix)
```

Comment:
Same as above: `append` mutates the argument, so this would effect the calling code. The source argument should be a copy:

```suggestion
	keysPath := append(path[:], dictionaryKeyPathPrefix)
```

Reply:
Resolving as per discussion below